### PR TITLE
UI/15399704 modal base

### PR DESCRIPTION
**Pull request by KatarzynaStefanowiczPagepro**
## Description

- [] TODO write description

**New stuff** ✨

* New stuff goes here

**Changes** 🏗

* Changes go here

**Deletions** ⚰️

* Deletions go here

## TODO

- [ ] Add TODOs

(Resolves | Contributes to) #31415

### CODE REVIEW

File: src/components/shared/Extensions/Modal/ModalBase.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import React, { FC } from 'react';
+import ReactModal from 'react-modal';
+import clsx from 'clsx';
+import styles from './Modal.module.css';
```
Comment (agnieszkajarosikloj) [inline src/components/shared/Extensions/Modal/ModalBase.tsx:4]: Style import at the end

**Comment by KatarzynaStefanowiczPagepro** [comment]
https://tw.auroracreation.com/app/tasks/15399704

File: src/components/shared/Extensions/Modal/ModalBase.tsx (hunk 1 above)
Comment (KatarzynaStefanowiczPagepro) [reply src/components/shared/Extensions/Modal/ModalBase.tsx]: Fixed

File: src/components/shared/Extensions/Modal/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import React, { FC, PropsWithChildren } from 'react';
+import { useIntl } from 'react-intl';
+import Icon from '~shared/Icon';
+import styles from './Modal.module.css';
```
[the remark this answers is not in the document]
Comment (KatarzynaStefanowiczPagepro) [reply src/components/shared/Extensions/Modal/Modal.tsx]: Fixed

**Review by agnieszkajarosikloj — APPROVED** [review]

File: src/components/shared/Extensions/Modal/Modal.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  const { formatMessage } = useIntl();
+
+  const titleText = typeof title == 'string' ? title : title && formatMessage(title);
+
+  return (
+    <ModalBase onRequestClose={onClose} {...props}>
+      <div className="flex mb-4 relative">
+        {icon && (
+          <span className={styles.icon}>
+            {/* @TODO when icons will work */}
```
Comment (krawczykkatarzyna) [inline src/components/shared/Extensions/Modal/Modal.tsx:20]: is the comment necessary? you added icon below

**Pull request merged**